globals.d.ts

```diff
@@ -52,6 +52,137 @@ export interface DatabaseObject {
   [key: string]: unknown;
 }
 
+/**
+ * Search result object; similar to `DatabaseObject` but without some properties.
+ */
+export type SearchResultsObject =
+  | SearchResultsObjectProps
+  | SearchResultsObjectGenerics;
+
+interface SearchResultsObjectProps {
+  "@id": string;
+  "@type": string[];
+}
+
+interface SearchResultsObjectGenerics {
+  [key: string]: unknown;
+}
+
+/**
+ * The `properties` object of a schema.
+ */
+export type SchemaProperties = SchemaPropertiesProps | SchemaPropertiesGenerics;
+
+export interface SchemaPropertiesProps {
+  title: string;
+  type: string;
+}
+
+export interface SchemaPropertiesGenerics {
+  [key: string]: unknown;
+}
+
+/**
+ * Describes schema objects.
+ */
+export interface Schema {
+  $id: string;
+  $schema: string;
+  "@type": string[];
+  additionalProperties: boolean;
+  changelog?: string;
+  dependentSchemas?: {
+    [key: string]: {
+      [key: string]: unknown;
+    };
+  };
+  description?: string;
+  exact_searchable_fields?: string[];
+  fuzzy_searchable_fields?: string[];
+  identifyingProperties?: string[];
+  mixinProperties: object[];
+  properties: SchemaProperties;
```

Comment:
Maybe I'm missing something, but I think this might be off? When I'm looking at the `properties` field of a schema, I'm seeing:
```
prop_name: {
  "title": string,
  "type": string,
   "description": string,
   // etc
}
```
So I think for `properties` you'd want something closer to:
```
// ...
properties:   [key: string]: SchemaProperties;
// ...
```